packages/langium/test/grammar/type-system/type-validator.test.ts

```diff
@@ -78,4 +78,20 @@ describe('validate params in types', () => {
         expect(diagnostics.filter(d => d.severity === DiagnosticSeverity.Error)).toHaveLength(0);
     });
 
-});
\ No newline at end of file
+});
+
+describe('validate inferred types', () => {
+
+    test('inferred type in cross-reference should not produce an error', async () => {
+        const prog = `
+        A infers B: 'a' name=ID (otherA=[B])?;
+        hidden terminal WS: /\\s+/;
+        terminal ID: /[a-zA-Z_][a-zA-Z0-9_]*/;
+        `.trim();
+
+        const document = await parseDocument(grammarServices, prog);
+        const diagnostics: Diagnostic[] = await grammarServices.validation.DocumentValidator.validateDocument(document);
+        expect(diagnostics.filter(d => d.severity === DiagnosticSeverity.Error)).toHaveLength(0);
+
+    });
+});
```

Comment:
Could you add a newline?

Reply:
The diff isn't showing it, but there is a newline present at the end 👍 .